internal/commands/result_test.go

```diff
@@ -132,6 +133,192 @@ func TestResultsExitCode_OnPartialScan_PrintOnlyFailedScannersInfoToConsole(t *t
+		if result.Type == params.SCSSecretDetectionType {
+			scsSecretDetectionFound = true
+		}
+		if result.Type == params.SCSScorecardType {
+			scsScorecardFound = true
+		}
+		if scsSecretDetectionFound && scsScorecardFound {
+			break
+		}
+	}
+	assert.Assert(t, scsSecretDetectionFound && scsScorecardFound, "SCS results should be included for AST-CLI agent")
+	assert.Assert(t, results.TotalCount == 2, "SCS Scorecard results should be excluded for VS Code agent")
+
+	defer os.Remove(fileName + ".json")
+}
+
+func TestRunScsResultsShow_VSCode_AgentShouldNotShowScorecardResults(t *testing.T) {
+	executeCommand := func(agent string) *wrappers.ScanResultsCollection {
+		clearFlags()
+		mock.Flag = wrappers.FeatureFlagResponseModel{Name: wrappers.SCSEngineCLIEnabled, Status: true}
+
+		_, err := executeRedirectedOsStdoutTestCommand(createASTTestCommand(),
+			"results", "show", "--scan-id", "SCS", "--report-format", "json", "--agent", agent)
+		assert.NilError(t, err)
+
+		file, err := os.Open(fileName + ".json")
+		if err != nil {
+			t.Fatalf("failed to open file: %v", err)
+		}
+		defer file.Close()
+
+		fileContents, err := io.ReadAll(file)
+		if err != nil {
+			t.Fatalf("failed to read file: %v", err)
+		}
+
+		var results wrappers.ScanResultsCollection
+		err = json.Unmarshal(fileContents, &results)
+		assert.NilError(t, err)
+		return &results
+	}
+
+	results := executeCommand(params.VSCodeAgent)
+	for _, result := range results.Results {
+		assert.Assert(t, result.Type != params.SCSScorecardType, "SCS Scorecard results should be excluded for VS Code agent")
+	}
+	assert.Assert(t, results.TotalCount == 1, "SCS Scorecard results should be excluded for VS Code agent")
+
+	defer os.Remove(fileName + ".json")
+}
+
+func TestRunScsResultsShow_Other_AgentsShouldNotShowScsResults(t *testing.T) {
+	executeCommand := func(agent string) *wrappers.ScanResultsCollection {
+		clearFlags()
+		mock.Flag = wrappers.FeatureFlagResponseModel{Name: wrappers.SCSEngineCLIEnabled, Status: true}
+
+		_, err := executeRedirectedOsStdoutTestCommand(createASTTestCommand(),
+			"results", "show", "--scan-id", "SCS", "--report-format", "json", "--agent", agent)
+		assert.NilError(t, err)
+
+		file, err := os.Open(fileName + ".json")
+		if err != nil {
+			t.Fatalf("failed to open file: %v", err)
+		}
+		defer file.Close()
+
+		fileContents, err := io.ReadAll(file)
+		if err != nil {
+			t.Fatalf("failed to read file: %v", err)
+		}
+
+		var results wrappers.ScanResultsCollection
+		err = json.Unmarshal(fileContents, &results)
+		assert.NilError(t, err)
+		return &results
+	}
+
+	results := executeCommand("Jetbrains")
+	for _, result := range results.Results {
+		assert.Assert(t, result.Type != params.SCSScorecardType && result.Type != params.SCSSecretDetectionType, "SCS results should be excluded for other agents")
+	}
+	assert.Assert(t, results.TotalCount == 0, "SCS Scorecard results should be excluded")
+
+	defer os.Remove(fileName + ".json")
```

Comment:
and here

Reply:
just call it without the defer. the defer is for setting the object for null ( dispose) but not for our own logic